File: lib_audio/src/main/java/com/imooc/lib_audio/mediaplayer/view/adapter/MusicPagerAdapter.java
package com.imooc.lib_audio.mediaplayer.view.adapter;

import android.animation.ObjectAnimator;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.view.PagerAdapter;
import android.util.SparseArray;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.LinearInterpolator;
import android.widget.ImageView;

import com.imooc.lib_audio.R;
import com.imooc.lib_audio.mediaplayer.core.AudioController;
import com.imooc.lib_audio.mediaplayer.model.AudioBean;
import com.imooc.lib_image_loader.app.ImageLoaderManager;

import java.util.ArrayList;

public class MusicPagerAdapter extends PagerAdapter {

	private ArrayList<AudioBean> mData;
	private Context mContext;


	private SparseArray<ObjectAnimator> mAnims = new SparseArray<>();
	public MusicPagerAdapter(ArrayList<AudioBean> queue, Context context, Object o) {
		mData = queue;
		mContext = context;
	}

	@NonNull
	@Override
	public Object instantiateItem(@NonNull ViewGroup container, int position) {
		View rootView = LayoutInflater.from(mContext).inflate(R.layout.indictor_item_view, null);
		//显示专辑图片
		final ImageView imageView = rootView.findViewById(R.id.circle_view);
		container.addView(rootView);
		ImageLoaderManager.getInstance()
				.displayImageForCircle(imageView, mData.get(position).getAlbumPic());
		mAnims.put(position, createAnim(imageView));
		return rootView;
	}

	private ObjectAnimator createAnim(ImageView imageView) {
		imageView.setRotation(0);
		ObjectAnimator animator = ObjectAnimator.ofFloat(imageView, View.ROTATION.getName(), 0, 360);
		animator.setDuration(10000);
		animator.setInterpolator(new LinearInterpolator());
		animator.setRepeatCount(-1);
		if (AudioController.INSTANCE.isStartState()) {
			animator.start();
		}
		return animator;
	}



	public ObjectAnimator getAnims(int position){
		return mAnims.get(position);
	}

	@Override
	public int getCount() {
		return mData == null ? 0 : mData.size();
	}

	@Override
	public boolean isViewFromObject(@NonNull View view, @NonNull Object o) {
		return view == o;
	}

	@Override
	public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
		container.removeView((View) object);
	}
}

File: lib_audio/src/main/java/com/imooc/lib_audio/mediaplayer/view/adapter/MusicPagerAdapter.kt
package com.imooc.lib_audio.mediaplayer.view.adapter

import android.animation.ObjectAnimator
import android.content.Context
import android.support.v4.view.PagerAdapter
import android.util.SparseArray
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.LinearInterpolator
import android.widget.ImageView
import com.imooc.lib_audio.R
import com.imooc.lib_audio.mediaplayer.core.AudioController
import com.imooc.lib_audio.mediaplayer.model.AudioBean
import com.imooc.lib_image_loader.app.ImageLoaderManager

class MusicPagerAdapter(queue: ArrayList<AudioBean>, context: Context) : PagerAdapter() {

    private var mData: ArrayList<AudioBean> = queue
    private var mContext: Context = context

    private val mAnims: SparseArray<ObjectAnimator> = SparseArray()

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val rootView: View = LayoutInflater.from(mContext).inflate(R.layout.indictor_item_view, null)
        val imageView: ImageView = rootView.findViewById(R.id.circle_view);
        container.addView(rootView)
        ImageLoaderManager.getInstance().displayImageForCircle(imageView, mData[position].albumPic)
        mAnims.put(position, createAnim(imageView))
        return super.instantiateItem(container, position)
    }


    private fun createAnim(imageView: ImageView): ObjectAnimator{
        imageView.rotation = 0f
        val animator: ObjectAnimator = ObjectAnimator.ofFloat(imageView, View.ROTATION.name, 0f, 360f)
        animator.duration = 10000
        animator.interpolator = LinearInterpolator()
        animator.repeatCount = -1
        if(AudioController.isStartState()){
            animator.start()
        }
        return animator
    }

    fun getAnims(position: Int): ObjectAnimator? {
        return mAnims[position]
    }

    override fun isViewFromObject(view: View, o: Any): Boolean {
        return view == o
    }

    override fun getCount(): Int {
       return mData.size
    }
}